pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -50,17 +53,26 @@ type AbstractConnectionManager struct {
 	Configuration        *ManagerConfigration
 
 	controlMessageSocket *ControlMessageSocket
+	allocator            eventprocessor.Allocator
 }
 
 func NewAbstractConnectionManager(parentLogger logger.Logger, runtimeConfiguration runtime.Configuration, configuration *ManagerConfigration) *AbstractConnectionManager {
 	// TODO: make MinConnectionsNum and MaxConnectionsNum configurable when support multiple event connections
-	return &AbstractConnectionManager{
+	abstractConnectionManager := &AbstractConnectionManager{
 		Logger:               parentLogger.GetChild("connection-manager"),
 		MinConnectionsNum:    1,
 		MaxConnectionsNum:    1,
 		RuntimeConfiguration: runtimeConfiguration,
 		Configuration:        configuration,
 	}
+	if abstractConnectionManager.MinConnectionsNum == 1 && abstractConnectionManager.MaxConnectionsNum == 1 {
+		// add support sync singleton
+		abstractConnectionManager.allocator = eventprocessor.NewSingletonAllocator(abstractConnectionManager.Logger, nil)
```

Comment:
Why not use sync pool here too?

Reply:
@TomerShor 1- we don't have a sync pool implementation yet, 2 - it is not needed if worker allocator is sync. So it is either todo or just something we should cover with validation (sooner the 2nd option)